### PR TITLE
feat: adding `Career` link to user menu

### PR DESCRIPTION
**Pull request by krishatcher**
### Description:

Adding a new link to the User Menu for logged-in users. This will redirect the user to the "Career Resource Center" or "CRC".

**GO-LIVE DATE CHANGE**
**Note that this PR should not go to Production until on or after 6 June 2023, in preparation for the launch of the CRC.**

### JIRA:

[CAMS-1](https://2u-internal.atlassian.net/browse/CAMS-1)

### CODE REVIEW

File: src/Header.jsx [hunk 1]
```diff
@@ -91,8 +91,20 @@ const Header = ({ intl }) => {
     baseUserMenuDashboardLinks = [dashboardMenuItem];
   }
 
+  const careerItemContent = `${intl.formatMessage(messages['header.user.menu.career'])} <span className="badge badge-warning">${intl.formatMessage(messages['header.user.menu.newAlert'])}</span>`;
```
Comment (Syed-Ali-Abbas-Zaidi) [inline src/Header.jsx:94]: `Career New` is not appearing as per the design given in [CAMS-1](https://2u-internal.atlassian.net/browse/CAMS-1).

Screenshot 2023-05-19 at 11 10 58 AM

**Review by Syed-Ali-Abbas-Zaidi — CHANGES_REQUESTED** [review]

File: src/Header.jsx (hunk 1 above)
Comment (leangseu-edx) [reply src/Header.jsx:94]:
```suggestion
  const careerItemContent = (${intl.formatMessage(messages['header.user.menu.career'])}<span className="badge badge-warning">${intl.formatMessage(messages['header.user.menu.newAlert'])}</span>);
```

Comment (Syed-Ali-Abbas-Zaidi) [reply src/Header.jsx]: Hi @krishatcher, As we have removed the [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) we don't need the `$` sign anymore. That is one of the reasons that lint is failing. You can replace it with the given code snippet:

`const careerItemContent = <>{intl.formatMessage(messages['header.user.menu.career'])}<span className="badge badge-warning">{intl.formatMessage(messages['header.user.menu.newAlert'])}</span></>;`

By utilizing the aforementioned code snippet, the rendering of `Career New` is now being executed correctly. However, the overall design still appears to be lacking.

Screenshot 2023-05-20 at 2 01 38 AM

File: src/Header.jsx [hunk 2]
```diff
@@ -91,8 +92,20 @@ const Header = ({ intl }) => {
     baseUserMenuDashboardLinks = [dashboardMenuItem];
   }
 
+  const careerItemContent = <>{intl.formatMessage(messages['header.user.menu.career'])}&nbsp;&nbsp;<Badge className="px-2 mx-2" variant="warning">{intl.formatMessage(messages['header.user.menu.newAlert'])}</Badge></>;
```
Comment (Syed-Ali-Abbas-Zaidi) [inline src/Header.jsx:95]: Currently, the badge is appearing as a rectangle. There is a boolean prop `pill` in the `Badge` component that will make it look like a pill.

`<Badge className="px-2 mx-2" variant="warning" pill>`

Screenshot 2023-05-23 at 7 13 13 PM

Adding a `pill` prop will make it look exactly like the design given in the ticket.

Screenshot 2023-05-23 at 7 16 48 PM

File: src/learning-header/AuthenticatedUserDropdown.jsx [hunk 3]
```diff
@@ -16,6 +16,14 @@ const AuthenticatedUserDropdown = ({ enterpriseLearnerPortalLink, intl, username
       {intl.formatMessage(messages.dashboard)}
     </Dropdown.Item>
   );
+  let careersMenuItem = (
+    <Dropdown.Item href="https://careers.edx.org/">
+      {intl.formatMessage(messages.career)}
+      <Badge className="px-2 mx-2" variant="warning">
```
Comment (Syed-Ali-Abbas-Zaidi) [inline src/learning-header/AuthenticatedUserDropdown.jsx:22]: We should add the `pill` prop over here too.

`<Badge className="px-2 mx-2" variant="warning" pill>`

File: src/Header.jsx (hunk 2 above)
Comment (Syed-Ali-Abbas-Zaidi) [inline src/Header.jsx:95]: We can also remove both `&nbsp;` as the space between Career and New is already being handled in classname `px-2 mx-2`.

**Review by Syed-Ali-Abbas-Zaidi — CHANGES_REQUESTED** [review]

File: src/Header.jsx (hunk 2 above)
Comment (krishatcher) [reply src/Header.jsx:95]: It's showing the rounded rectangle elsewhere, so I'd like to leave it that way here too.
Desktop Screen Shot

Comment (krishatcher) [reply src/Header.jsx]: non-breaking space chars removed

**Review by Syed-Ali-Abbas-Zaidi — APPROVED** [review]
LGTM.

**Comment by krishatcher** [comment]
This PR is part of a coordinated effort across multiple edX properties.  The full list of related PRs are included below for reference:

* https://github.com/edx/edx-themes/pull/794
* https://github.com/openedx/edx-platform/pull/32296
* https://github.com/edx/prospectus/pull/5345
* https://github.com/openedx/frontend-app-learner-dashboard/pull/152
* https://github.com/edx/edx-internal/pull/8428
* https://github.com/edx/frontend-component-header-edx/pull/375

**Pull request merged**